lib/ddtrace/context.rb

```diff
@@ -15,18 +15,45 @@ module Datadog
   # This data structure is thread-safe.
   class Context
     # Initialize a new thread-safe \Context.
-    def initialize
+    def initialize(options = {})
       @mutex = Mutex.new
-      reset
+      reset(options)
     end
 
-    def reset
+    def reset(options = {})
       @trace = []
-      @sampled = false
+      @parent_trace_id = options.fetch(:trace_id, nil)
+      @parent_span_id = options.fetch(:span_id, nil)
+      @sampled = options.fetch(:sampled, false)
```

Comment:
I'm mixed here. Python implementation uses `true` by default. We could change this, it possibly breaks some internal tests, but none of them should impact behavior from a user POV. AFAIK client sampling is not really massively used today in Ruby, and as a side note, first span added would turn this to `true`. But I think it's worth noticing.